assets/ios/NotificationService.m

```diff
@@ -135,7 +142,64 @@ - (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withConte
         self.bestAttemptContent.body = [NSString stringWithFormat:@"You received %.0f sats ⚡️", amountInSats];
     }
 
-    self.contentHandler(self.bestAttemptContent);
+    if ([notificationType isEqualToString:@"payment_received"] && ttsEnabled) {
+        NSString *speechText         = [NSString stringWithFormat:@"%.0f sats", amountInSats];
+        self.synthesizer             = [[AVSpeechSynthesizer alloc] init];
+        self.synthesizer.delegate    = self;
+        AVSpeechUtterance *utterance = [[AVSpeechUtterance alloc] initWithString:speechText];
+        __block AVAudioFile *audioFile;
+        NSString *audioName     = @"tts.caf";
+        NSFileManager *fileMan  = [NSFileManager defaultManager];
+        NSString *containerPath = [fileMan containerURLForSecurityApplicationGroupIdentifier:@"group.com.getalby.mobile.nse"].path;
+        NSString *soundDir = [containerPath stringByAppendingString:@"/Library/Sounds"];
+        NSError *errorOut;
+        if (![fileMan fileExistsAtPath:soundDir]) {
+            [fileMan createDirectoryAtPath:soundDir withIntermediateDirectories:YES attributes:nil error:&errorOut];
+        } else {
+            // Delete old files
+            NSArray<NSString*> *oldFiles = [fileMan contentsOfDirectoryAtPath:soundDir error:&errorOut];
+            if (oldFiles != nil) {
+                for (NSString *oldFileName in oldFiles) {
+                    NSString *oldFilePath = [soundDir stringByAppendingPathComponent:oldFileName];
+                    [fileMan removeItemAtPath:oldFilePath error:&errorOut];
+                }
+            }
+        }
+        if (errorOut != nil) {
+          return;
+        }
+        NSString *soundPath = [soundDir stringByAppendingString:[@"/" stringByAppendingString:audioName]];
+        NSURL *soundUrl     = [NSURL fileURLWithPath:soundPath];
+        [self.synthesizer writeUtterance:utterance toBufferCallback:^(AVAudioBuffer * _Nonnull buffer) {
+            if (![buffer isKindOfClass:[AVAudioPCMBuffer class]]) {
+              return;
+            }
+            AVAudioPCMBuffer *pcmBuffer = (AVAudioPCMBuffer *)buffer;
+            if (pcmBuffer.frameLength == 0) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                  self.bestAttemptContent.sound = [UNNotificationSound soundNamed:audioName];
+                  contentHandler(self.bestAttemptContent);
+                  self.synthesizer = nil;
+                });
+                return;
+            } else {
+                if (!audioFile) {
```

Comment:
I don't understand the content of this `else`, it's writing the file but then what happens with it? where does the notification use it?